native/src/main/java/io/ballerina/lib/avro/visitor/SerializeVisitor.java

```diff
@@ -0,0 +1,348 @@
+        Arrays.stream(data.getValues())
+                .filter(Objects::nonNull)
+                .forEach(record -> {
+                    try {
+                        array.add(new SerializeVisitor().visitMap((BMap<?, ?>) record, schema.getElementType()));
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+        return array;
+    }
+
+    private GenericData.Array<Object> visitBytes(BArray data, GenericData.Array<Object> array) {
+        Arrays.stream(data.getValues())
+                .filter(Objects::nonNull)
+                .forEach(bytes -> {
+                    ByteBuffer byteBuffer = ByteBuffer.allocate(((BArray) bytes).getByteArray().length);
+                    byteBuffer.put(((BArray) bytes).getByteArray());
+                    byteBuffer.position(0);
+                    array.add(byteBuffer);
+                });
+        return array;
+    }
+
+    private GenericData.Array<Object> visitFixed(BArray data, GenericData.Array<Object> array, Schema schema) {
+        Arrays.stream(data.getValues())
+                .filter(Objects::nonNull)
+                .forEach(bytes -> {
+                    GenericFixed genericFixed = new GenericData.Fixed(schema, ((BArray) bytes).getByteArray());
+                    array.add(genericFixed);
+                });
+        return array;
+    }
+
+    private GenericData.Array<Object> visitBooleanArray(BArray data, GenericData.Array<Object> array) {
+        for (Object obj: data.getBooleanArray()) {
+            array.add(obj);
+        }
+        return array;
+    }
+
+    private GenericData.Array<Object> visitDoubleArray(BArray data, GenericData.Array<Object> array) throws Exception {
+        try {
+            for (Object obj: data.getFloatArray()) {
+                array.add(obj);
+            }
+            return array;
+        } catch (Exception e) {
+            throw new Exception(e.getMessage());
+        }
+    }
+
+    private GenericData.Array<Object> visitLongArray(BArray data, GenericData.Array<Object> array) {
+        for (Object obj: data.getIntArray()) {
+            array.add(obj);
+        }
+        return array;
+    }
+
+    private GenericData.Array<Object> visitStringArray(BArray data, GenericData.Array<Object> array) {
+        array.addAll(Arrays.asList(data.getStringArray()));
+        return array;
+    }
+
+    private Object visitUnion(Object data, Schema.Field field) throws Exception {
+        Type typeName = TypeUtils.getType(data);
+        switch (typeName.getClass().getSimpleName()) {
+            case STRING_TYPE -> {
+                for (Schema schema : field.schema().getTypes()) {
+                    if (schema.getType().equals(Schema.Type.ENUM)) {
+                        return new GenericData.EnumSymbol(schema, data);
+                    }
+                }
```

Comment:
Better to move this to a separate method. This loop makes the switch statement hard to read and increases the cognitive complexity.